staging/src/k8s.io/client-go/discovery/fake/discovery.go

```diff
@@ -67,7 +80,11 @@ func (c *FakeDiscovery) ServerPreferredNamespacedResources() ([]*metav1.APIResou
 }
 
 func (c *FakeDiscovery) ServerGroups() (*metav1.APIGroupList, error) {
-	return nil, nil
+	return c.serverGroups, nil
```

Comment:
This appears to be building the layer the wrong location.  It seems like you want a simple skin on the `*testing.Fake` which setups up the "proper" handling for the various `Invokes` calls by the  `PrependReactor` methods.  That way you can use the normal check of `Actions` if you need to.

I'm fine with making it easier to set those, but I'd like to see these calls making their proper remote calls.  That may require refactoring the `DiscoveryClient` itself to separate the accessors from the helpers, but I think that is more productive than doing it  like this.